src/gluonts/mx/trainer/callback.py

```diff
@@ -338,3 +341,39 @@ def on_network_initializing_end(
         self, training_network: nn.HybridBlock
     ) -> None:
         copy_parameters(self.predictor.prediction_net, training_network)
+
+
+class TrainingTimeLimitCallback(Callback):
+    def __init__(self, time_limit=None):
+        """
+        Used when you want to set a time limit to the training process.
+        Once passed into model.train(), the training process will end roughly after 'time_limit' seconds.
```

Comment:
Can we a bit more precise? Why is this "roughly"?

Reply:
The actual training time might be a little bit over `time_limit`. For example, if `time_limit` is 60s, the first 10 epochs take 55s, and the 11st epoch takes 10s, the training process will stop after epoch 11, which takes 65s in total.

Reply:
Could we add this to the documentation? Something like "the training process will end after the first batch exceeding `time_limit` seconds"?

Reply:
Ah it's actually epoch, not batch.